huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_catalog.go

```diff
@@ -0,0 +1,365 @@
+		},
+		JSONBody: map[string]interface{}{
+			"target_pid": d.Get("parent_id"),
+		},
+	}
+
+	_, err := client.Request("POST", createPath, &opt)
+	if err != nil {
+		return fmt.Errorf("error moving current catalog to a new path: %s", err)
+	}
+	return nil
+}
+
+func resourceDatatServiceCatalogUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+	client, err := cfg.NewServiceClient("dataarts", region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	if d.HasChanges("name", "description") {
+		err = updateCatalogBasicConfigs(client, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if d.HasChange("parent_id") {
+		err = updateCatalogParentId(client, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceDatatServiceCatalogRead(ctx, d, meta)
+}
+
+func resourceDatatServiceCatalogDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		httpUrl     = "v1/{project_id}/service/servicecatalogs/batch-delete"
+		workspaceId = d.Get("workspace_id").(string)
+		catalogId   = d.Id()
+	)
+	client, err := cfg.NewServiceClient("dataarts", region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+	// Due to the lack of an interface for deleting a single directory, the corresponding function can only be
+	// implemented through the batch deletion interface, and the batch deletion API cannot be sent to the server
+	// at the same time, which will cause an error to be returned.
+	config.MutexKV.Lock(workspaceId)
+	defer config.MutexKV.Unlock(workspaceId)
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes:          []int{204},
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+			"workspace":    workspaceId,
+			"Dlm-Type":     d.Get("dlm_type").(string),
+		},
+		JSONBody: map[string]interface{}{
+			"ids": []string{catalogId},
+		},
+	}
+
+	_, err = client.Request("POST", createPath, &opt)
+	if err != nil {
+		return diag.Errorf("error deleting catalog (%s): %s", catalogId, err)
+	}
+	return nil
+}
+
+func resourceDataServiceCatalogImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	importedId := d.Id()
+	parts := strings.Split(importedId, "/")
+	if len(parts) != 2 && len(parts) != 3 {
+		return nil, fmt.Errorf("invalid format specified for import ID, must be '<workspace_id>/<dlm_type>/<id>' or "+
+			"'<workspace_id>/<id>', but got '%s'", importedId)
+	}
+
+	d.Set("workspace_id", parts[0])
+	if len(parts) == 2 {
+		d.SetId(parts[1])
+	}
+	if len(parts) == 3 {
+		d.Set("dlm_type", parts[1])
```

Comment:
Please handle possible errors in set `workspace_id` and `dlm_type`.

Reply:
done.